server/sonar-process-monitor/src/test/java/org/sonar/application/cluster/AppStateClusterImplTest.java

```diff
@@ -116,6 +121,41 @@ public void simulate_network_cluster() throws InterruptedException {
     }
   }
 
+  @Test
+  public void appstateclusterimpl_must_set_sonarqube_version() {
+    TestAppSettings settings = newClusterSettings();
+
+    try (AppStateClusterImpl appStateCluster = new AppStateClusterImpl(settings)) {
+      appStateCluster.registerSonarQubeVersion(getSonarqubeVersion());
+
+      HazelcastInstance hzInstance = createHazelcastClient(appStateCluster);
+      assertThat(hzInstance.getAtomicReference(SONARQUBE_VERSION).get())
+        .isNotNull()
+        .isInstanceOf(String.class)
+        .isEqualTo(getSonarqubeVersion());
+    }
+  }
+
+  @Test
+  public void incorrect_sonarqube_version_must_trigger_an_exception() throws IOException, InterruptedException, IllegalAccessException, NoSuchFieldException {
+    // Now launch an instance that try to be part of the hzInstance cluster
+    TestAppSettings settings = new TestAppSettings();
+    settings.set(ProcessProperties.CLUSTER_ENABLED, "true");
+    settings.set(ProcessProperties.CLUSTER_NAME, "sonarqube");
+
+
+    try (AppStateClusterImpl appStateCluster = new AppStateClusterImpl(settings)) {
+      // Register first version
+      appStateCluster.registerSonarQubeVersion("1.0.0");
+
+      expectedException.expect(IllegalStateException.class);
+      expectedException.expectMessage(Matchers.matches("The local version .* is not the same as the cluster 1\\.0\\.0"));
```

Comment:
Why not simply `expectedException.expectMessage("The local version 2.0.0 is not the same as the cluster 1.0.0");` ?